package.json

```diff
@@ -40,18 +41,26 @@
     "turbo": "^2.2.3",
     "typesync": "0.13.0"
   },
-  "packageManager": "pnpm@9.15.1",
+  "packageManager": "pnpm@9.15.2",
   "engines": {
     "node": ">=20.10.0"
   },
   "pnpm": {
     "peerDependencyRules": {
-      "allowAny": [
-        "typescript"
-      ],
+      "allowedVersions": {
+        "eslint": "^9.0.0",
+        "typescript": "^5.0.0"
+      },
       "ignoreMissing": [
         "vitest"
       ]
-    }
+    },
+    "onlyBuiltDependencies": [
+      "@swc/core",
+      "core-js",
+      "core-js-pure",
+      "esbuild",
+      "sharp"
```

Comment:
i have no idea what this is. I looked up what `onlyBuiltDependencies` is, but i'm not sure why we would need this. Can you elaborate?

Reply:
pnpm v10 disable postinstall scripts from running as a security measure. This opts us in to that behavior both to future-proof and to improve our security posture.
This is saying that only these packages can run postinstalls, which was the status quo (perhaps we should disable core-js?).

Reply:
I re-installed the deps in my local env, and as far as i can tell only `swc/core`  run postinstall scripts. Why the other deps? Are you seeing something different?

Reply:
pnpm uses a "side-effects cache" which saves the postinstalls so that stuff like sharp's postinstall don't have to run every time (it can take 2 minutes on my (not particularly beefy) laptop).

Reply:
Thank you for the explanation.

I re-run the install with the cache turned off and now i can see the post-install 👍